drivers/gpio/gpio_nrfe.c

```diff
@@ -127,7 +158,7 @@ static int gpio_nrfe_init(const struct device *port)
 	}
 
 	k_sem_take(&bound_sem, K_FOREVER);
-
+#endif
```

Comment:
I am not sure if leaving init function for mbox is a good idea. Mbox should most likely be verified, if it bonded with the other core. Right now ARM can send messages to FLPR before it enables its mbox. As a result, some of the messages will disappear, since sram_tx is overwritten every time a new message is "sent" (either a circular buffer or a separate inter-core semaphore would be useful here to make sure that the message was read). It will make no difference in blinky, but it will matter in case other eperipherals. Unless I misunderstood something.
I am not saying it should be added to this PR, we can add it later, but it would be nice if it was. I'm afraid we will forget about it otherwise.

Reply:
Ok, you're right, I haven't noticed clearing this atomic flag in the main of FLPR code. However, I still have some concerns regarding synchronization. The current mechanism will prevent APP from writing before FLPR has mbox initialized, but APP still does not know if the FLPR has read the message, because later this atomic variable is just used as a shared memory access semaphore/barrier (each core takes it before writing/reading and clears it right after). However, there is no signalization indicating that the other core has read the message and that the buffer can be overwritten unless I missed something again. It will most probably not be a problem here as the APP side has a delay before changing the state of GPIO again, so you don't have to change it in this PR. But it is something that we at least have to keep in mind when adding support for more complicated peripherals, for example, QSPI.

Reply:
Ok, you're right, I haven't noticed clearing this atomic flag in the main of FLPR code. However, I still have some concerns regarding synchronization. The current mechanism will prevent APP from writing before FLPR has mbox initialized, but APP still does not know if the FLPR has read the message, because later this atomic variable is just used as a shared memory access semaphore/barrier (each core takes it before writing/reading and clears it right after). However, there is no signalization indicating that the other core has read the message and that the buffer can be overwritten unless I missed something again. It will most probably not be a problem here as the APP side has a delay before changing the state of GPIO again, so you don't have to change it in this PR. But it is something that we at least have to keep in mind when adding support for more complicated peripherals, for example, QSPI.